app/assets/javascripts/admin/addon/components/modal/scrub-rejected-user.gjs

```diff
@@ -0,0 +1,67 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { isEmpty } from "@ember/utils";
+import DButton from "discourse/components/d-button";
+import DModal from "discourse/components/d-modal";
+import TextField from "discourse/components/text-field";
+import { i18n } from "discourse-i18n";
+
+export default class ScrubRejectedUserModal extends Component {
+  @tracked isScrubbing = false;
+  @tracked scrubReason = "";
+
+  @action
+  updateScrubReason(event) {
+    this.scrubReason = event.target.value;
+  }
+
+  @action
+  async confirmScrub() {
+    this.isScrubbing = true;
+    await this.args.model.confirmScrub(this.scrubReason);
+    this.args.closeModal();
+  }
+
+  get scrubButtonDisabled() {
+    return this.isScrubbing || isEmpty(this.scrubReason);
+  }
+
+  <template>
+    <DModal
+      @bodyClass="scrub-rejected-user"
+      class="admin-scrub-rejected-user-modal"
+      @title={{i18n "review.user.scrub_record.confirm_title"}}
+      @closeModal={{if this.isScrubbing null @closeModal}}
+    >
+      <:body>
+        <p>{{i18n "review.user.scrub_record.confirm_body"}}</p>
+        <label class="scrub-reason-title" for="scrub-reason">{{i18n
+            "review.user.scrub_record.reason_title"
+          }}</label>
+
+        <TextField
+          class="scrub-reason"
+          id="scrub-reason"
+          @placeholderKey="review.user.scrub_record.reason_placeholder"
+          {{on "input" this.updateScrubReason}}
```

Comment:
I think that if you add `@value={{this.scrubReason}}` `updateScrubReason` will be not necessary

Reply:
I tried doing that originally, but then typing in the field immediately throws a JS error:

`You attempted to update `attrs` on `<DiscourseTextField:ember173>`, but it had already been used previously in the same computation.  Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, and performance issues, and is not supported.`

Reply:
I just played with it, this is weird... 

Maybe at least make inline `mut`

```javascript
        <TextField
          class="scrub-reason"
          id="scrub-reason"
          @placeholderKey="review.user.scrub_record.reason_placeholder"
          {{on "input" (action (mut this.scrubReason) value="target.value")}}
        />
```